x-pack/plugins/infra/server/services/rules/types.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PluginSetupContract as AlertingPluginSetup } from '../../../../alerting/server';
+import {
+  createLifecycleExecutor,
+  RuleDataClient,
+  RuleRegistryPluginSetupContract,
+} from '../../../../rule_registry/server';
+
+type LifecycleRuleExecutorCreator = ReturnType<typeof createLifecycleExecutor>;
+
+export interface RulesServiceSetupDeps {
+  alerting: AlertingPluginSetup;
+  ruleRegistry: RuleRegistryPluginSetupContract;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RulesServiceStartDeps {}
+
+export interface RulesServiceSetup {
+  createLifecycleRuleExecutor: LifecycleRuleExecutorCreator;
+  ruleDataClient: RuleDataClient;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RulesServiceStart {}
+
+export type RuleRegistrationContext = 'observability.logs' | 'observability.metrics';
```

Comment:
I suggest we move hardcoded strings to the constants file I mentioned earlier

Reply:
I would agree in general, but looking at the discussions (such as #104958) I expect the way we register the alert data templates to change soon. Do you mind if I leave it here for now until we know what parameters the final rule data client API will require?

Reply:
Yep it makes sense. Let's leave it as it is for now